node/external/timemachine/fee/feeComputer.go

```diff
@@ -53,88 +45,30 @@ func NewFeeComputer(args ArgsNewFeeComputer) (*feeComputer, error) {
 
 // ComputeGasUsedAndFeeBasedOnRefundValue computes gas used and fee based on the refund value, at a given epoch
 func (computer *feeComputer) ComputeGasUsedAndFeeBasedOnRefundValue(tx *transaction.ApiTransactionResult, refundValue *big.Int) (uint64, *big.Int) {
-	instance, err := computer.getOrCreateInstance(tx.Epoch)
-	if err != nil {
-		log.Error("ComputeGasUsedAndFeeBasedOnRefundValue(): unexpected error when creating an economicsData instance", "epoch", tx.Epoch, "error", err)
-		return 0, big.NewInt(0)
-	}
-
-	return instance.ComputeGasUsedAndFeeBasedOnRefundValue(tx.Tx, refundValue)
+	return computer.economicsInstance.ComputeGasUsedAndFeeBasedOnRefundValueInEpoch(tx.Tx, refundValue, tx.Epoch)
 }
 
 // ComputeTxFeeBasedOnGasUsed computes fee based on gas used, at a given epoch
 func (computer *feeComputer) ComputeTxFeeBasedOnGasUsed(tx *transaction.ApiTransactionResult, gasUsed uint64) *big.Int {
-	instance, err := computer.getOrCreateInstance(tx.Epoch)
-	if err != nil {
-		log.Error("ComputeTxFeeBasedOnGasUsed(): unexpected error when creating an economicsData instance", "epoch", tx.Epoch, "error", err)
-		return big.NewInt(0)
-	}
-
-	return instance.ComputeTxFeeBasedOnGasUsed(tx.Tx, gasUsed)
+	return computer.economicsInstance.ComputeTxFeeBasedOnGasUsedInEpoch(tx.Tx, gasUsed, tx.Epoch)
 }
 
 // ComputeGasLimit computes a transaction gas limit, at a given epoch
 func (computer *feeComputer) ComputeGasLimit(tx *transaction.ApiTransactionResult) uint64 {
-	instance, err := computer.getOrCreateInstance(tx.Epoch)
-	if err != nil {
-		log.Error("ComputeGasLimit(): unexpected error when creating an economicsData instance", "epoch", tx.Epoch, "error", err)
-		return 0
-	}
-
-	return instance.ComputeGasLimit(tx.Tx)
+	return computer.economicsInstance.ComputeGasLimit(tx.Tx)
```

Comment:
you will probably need to add the epoch as well, check first my other comment in the economicsData.go